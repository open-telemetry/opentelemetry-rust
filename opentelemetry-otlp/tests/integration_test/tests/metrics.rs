use std::{fs::File, io::Write};

use integration_test_runner::metrics_asserter::{read_metrics_from_json, MetricsAsserter};
use opentelemetry_proto::tonic::metrics::v1::MetricsData;

#[test]
fn test_serde() {
    let metrics = read_metrics_from_json(File::open("./expected/metrics.json").unwrap());

    let json = serde_json::to_string_pretty(&MetricsData {
        resource_metrics: metrics,
    })
    .expect("Failed to serialize metrics");

    // Write to file.
    let mut file = File::create("./expected/serialized_metrics.json").unwrap();
    file.write_all(json.as_bytes()).unwrap();

    let left = read_metrics_from_json(File::open("./expected/metrics.json").unwrap());
    let right = read_metrics_from_json(File::open("./expected/serialized_metrics.json").unwrap());

    MetricsAsserter::new(left, right).assert();
}
