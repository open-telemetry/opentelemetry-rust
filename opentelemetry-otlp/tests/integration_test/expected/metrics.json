{
  "resourceMetrics": [
    {
      "resource": {
        "attributes": [
          {
            "key": "service.name",
            "value": {
              "stringValue": "metrics-integration-test"
            }
          }
        ]
      },
      "scopeMetrics": [
        {
          "scope": {
            "name": "meter"
          },
          "metrics": [
            {
              "name": "counter_u64",
              "sum": {
                "dataPoints": [
                  {
                    "attributes": [
                      {
                        "key": "mykey1",
                        "value": {
                          "stringValue": "myvalue1"
                        }
                      },
                      {
                        "key": "mykey2",
                        "value": {
                          "stringValue": "myvalue2"
                        }
                      }
                    ],
                    "startTimeUnixNano": "1734094309366798000",
                    "timeUnixNano": "1734094317871514000",
                    "asInt": "10"
                  }
                ],
                "aggregationTemporality": 2,
                "isMonotonic": true
              }
            },
            {
              "name": "example_histogram",
              "histogram": {
                "dataPoints": [
                  {
                    "attributes": [
                      {
                        "key": "mykey3",
                        "value": {
                          "stringValue": "myvalue4"
                        }
                      }
                    ],
                    "startTimeUnixNano": "1734094309366875000",
                    "timeUnixNano": "1734094317871537000",
                    "count": "1",
                    "sum": 42,
                    "bucketCounts": [
                      "0",
                      "0",
                      "0",
                      "0",
                      "1",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0",
                      "0"
                    ],
                    "explicitBounds": [
                      0,
                      5,
                      10,
                      25,
                      50,
                      75,
                      100,
                      250,
                      500,
                      750,
                      1000,
                      2500,
                      5000,
                      7500,
                      10000
                    ],
                    "min": 42,
                    "max": 42
                  }
                ],
                "aggregationTemporality": 2
              }
            },
            {
              "name": "example_up_down_counter",
              "sum": {
                "dataPoints": [
                  {
                    "attributes": [
                      {
                        "key": "mykey5",
                        "value": {
                          "stringValue": "myvalue5"
                        }
                      }
                    ],
                    "startTimeUnixNano": "1734094309366941000",
                    "timeUnixNano": "1734094317871548000",
                    "asInt": "-1"
                  }
                ],
                "aggregationTemporality": 2
              }
            }
          ]
        }
      ]
    }
  ]
}
